File: web/src/main/java/top/bettercode/summer/web/exception/ResourceNotFoundException.java
package top.bettercode.summer.web.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

/**
 * 找不到资源
 *
 * @author Peter Wu
 */
@ResponseStatus(HttpStatus.BAD_REQUEST)
public class ResourceNotFoundException extends RuntimeException {

  private static final long serialVersionUID = 1L;

  public ResourceNotFoundException() {
    this("resource.not.found");
  }

  public ResourceNotFoundException(String message) {
    this(message, null);
  }

  public ResourceNotFoundException(String message, Throwable cause) {
    super(message, cause);
  }
}

File: web/src/main/kotlin/top/bettercode/summer/web/exception/ResourceNotFoundException.kt
package top.bettercode.summer.web.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

/**
 * 找不到资源
 *
 * @author Peter Wu
 */
@ResponseStatus(HttpStatus.BAD_REQUEST)
class ResourceNotFoundException @JvmOverloads constructor(message: String? = "resource.not.found", cause: Throwable? = null) : RuntimeException(message, cause) {
    companion object {
        private const val serialVersionUID = 1L
    }
}